third_party/freedom-typings/freedom-common.d.ts

```diff
@@ -102,7 +113,8 @@ declare module freedom {
     // We use this specification so that you can reference freedom sub-modules by
     // an array-lookup of it's name. One day, maybe we'll have a nicer way to do
     // this.
-    [moduleName:string] : Function;
+    // TODO: explore how to use FreedomModuleFactoryManager.
+    [moduleName:string] : any;
```

Comment:
Ah, or does this relate to https://github.com/uProxy/uproxy/issues/852 ? I'm looking at that now, the vanilla JavaScript equivalent would be: https://github.com/freedomjs/freedom-pgp-e2e/blob/master/src/demo/e2edemo.js#L7

This file is the script for a freedom module that provides e2e as a dependency per this manifest: https://github.com/freedomjs/freedom-pgp-e2e/blob/master/src/demo/e2edemo.json

So if this corresponds to this typing then yes freedom can (from within its main module) instantiate submodules that it depends on. Anyway, will investigate further.